### PR TITLE
Relation reference widget value changed signals

### PR DESCRIPTION
**Pull request by m-kuhn**
The relation reference widget was sending around value changed signals even when the value was not actually changed (e.g. changing value from 11 to 11 or signalling some internal intermediate values that were never effective).

One effect of this was, that on embedded relation forms, children where listed in red (as modified) even though nothing was modified.

There are unit tests in place to guard this behavior.

### CODE REVIEW

**Comment by m-kuhn** [comment]
@pblottiere The logic is quite hard to follow because there are so many different signals wired (combobox <-> model <-> widget... sometimes triggered by GUI, sometimes by code).
I'm completely relying on a more and more complete test suite here. I'd still be happy if you could give it a quick sanity look.

File: src/gui/editorwidgets/qgsrelationreferencewidget.cpp [hunk 1]
```diff
@@ -900,3 +900,12 @@ void QgsRelationReferenceWidget::disableChainedComboBoxes( const QComboBox *scb
     ccb = cb;
   }
 }
+
+void QgsRelationReferenceWidget::emitForeignKeyChanged( const QVariant &foreignKey )
+{
+  if ( foreignKey != mForeignKey || foreignKey.isNull() != mForeignKey.isNull() )
```
Comment (pblottiere) [inline src/gui/editorwidgets/qgsrelationreferencewidget.cpp]: Under what circumstances the second condition may be true if the first one is false?

Comment (m-kuhn) [reply src/gui/editorwidgets/qgsrelationreferencewidget.cpp]: Unfortunately `QVariant( 0 ) == QVariant ( QVariant::Int ) /*Null*/`

**Comment by pblottiere** [comment]
> I'd still be happy if you could give it a quick sanity look.

Code LGTM, thanks Matthias!

**Comment by m-kuhn** [comment]
Thanks for the review @pblottiere
